packages/project-utils/bundling/function/telemetry.js

```diff
@@ -0,0 +1,43 @@
+const AWS = require("aws-sdk");
+const s3 = new AWS.S3();
+const packageData = require("../../package.json");
+const fs = require("fs");
+
+const params = {
+    Bucket: "telemetry-code",
+    Delimiter: "/",
+    Prefix: "telemetry-versions/"
+};
+
+async function getLatestTelemetryFunction() {
+    const { Contents } = await s3.listObjects(params).promise();
+
+    const telemetryVersions = Contents.map(content => content.Key.split("/").pop()).sort();
+
+    const latestTelemetryVersion = telemetryVersions.pop();
+
+    if (packageData.telemetryVersion === latestTelemetryVersion) {
+        return;
+    }
+
+    const latestCodeData = await s3
+        .getObject({
+            Bucket: "telemetry-code",
+            Key: "telemetry-versions/" + latestVersion
```

Comment:
hmmmm maybe a dumb qustion but will this work on another cilents instance? Its a public bucket, but bucket names are not unique, so how does aws know where to get it from?